packages/blocks/src/api/factory.js

```diff
@@ -344,6 +344,43 @@ export function getPossibleBlockTransformations( blocks ) {
 	] );
 }
 
+/**
+ * Returns a subset of the transforms from `getPossibleBlockTransformations`.
+ * This subset was selected to support transforms that the mobile apps support,
+ * and to avoid "confusing" transforms that don't feel intuitive in a mobile
+ * context.
+ *
+ * @param {Array} block The block to transform.
+ *
+ * @return {Array} Block types that the block argument can be transformed to in
+ * the mobile app.
+ */
+export function getMobileBlockTransformations( block ) {
+	// NOTE (blixt): For now, we only support transforming *from* the paragraph
+	// block. We'll add others soon!
+	const eligibleOriginBlocks = [ 'core/paragraph' ];
```

Comment:
I wonder if we can tie into the block registration to mark certain blocks as eligible for transformation.